aten/src/ATen/native/nested/NestedTensorMath.h

```diff
@@ -7,7 +7,32 @@ struct NestedTensorImpl;
 // TODO: cache this and only do it once per NestedTensor
 int64_t get_consistent_last_dim_of_nested_tensor(const NestedTensorImpl& nt);
 
-std::vector<int64_t> NestedTensor_get_max_size(const NestedTensorImpl& nt);
+inline std::vector<int64_t> NestedTensor_get_max_size_from_size_tensor(const Tensor& sizes) {
+  if (sizes.dim() == 0) {
+    return {};
+  }
+  const auto sizes_ptr = sizes.data_ptr<int64_t>();
+  const auto sizes_size_0 = sizes.sizes()[0];
+  const auto sizes_size_1 = sizes.sizes()[1];
+  TORCH_INTERNAL_ASSERT(sizes_size_1 > 0);
+  std::vector<int64_t> results(sizes_size_1, 0);
```

Comment:
shouldn't this be a DimVector? IIUC it's got expected length the same as sizes() of some Tensor. (we can make this a separate PR; I realize this is probably a move of stuff I did)

Reply:
Also, this seems like a pretty large function. Why is it inline in a header?

Reply:
I couldn't get it to link correctly if defined in NestedTensorMath.cpp, so I gave up and moved it into the header. Not an ideal solution.

Reply:
TORCH_API didn't help?